### PR TITLE
add option to create a registry pullsecret for openshift test

### PR DESCRIPTION
**Pull request by pkubatrh**
this pullsecret can be used to pull images from registries
that need authentication to access (eg. registry.redhat.io)

### CODE REVIEW

**Comment by pkubatrh** [comment]
@hhorak PTAL

**Comment by pkubatrh** [comment]
Actually might still need some work. Let me double check

**Comment by pkubatrh** [comment]
Fixed now.

This is needed for https://github.com/sclorg/redis-container/pull/53 but afaik we do not have any way to trigger openshift tests from common scripts in CI before merging. I verified this locally on a rhel7 VM.

File: test-lib-openshift.sh [hunk 1]
```diff
@@ -273,6 +273,13 @@ function ct_os_new_project() {
   oc new-project ${project_name}
   # let openshift cluster to sync to avoid some race condition errors
   sleep 3
+  if test -n "${OPENSHIFT_CLUSTER_PULLSECRET_PATH:-}" -a -e "${OPENSHIFT_CLUSTER_PULLSECRET_PATH:-}"; then
+    oc create -f "$OPENSHIFT_CLUSTER_PULLSECRET_PATH"
+    # add registry pullsecret to the serviceaccount if provided
+    secret_name=$(grep "name:" "$OPENSHIFT_CLUSTER_PULLSECRET_PATH" | awk '{ print $2 }')
```
Comment (hhorak) [inline test-lib-openshift.sh]: I'd suggest to use '^\s*name:' to avoid a potential wrong match in the future.

File: test-lib-openshift.sh [hunk 2]
```diff
@@ -273,6 +273,13 @@ function ct_os_new_project() {
   oc new-project ${project_name}
   # let openshift cluster to sync to avoid some race condition errors
   sleep 3
+  if test -n "${OPENSHIFT_CLUSTER_PULLSECRET_PATH:-}" -a -e "${OPENSHIFT_CLUSTER_PULLSECRET_PATH:-}"; then
```
Comment (hhorak) [inline test-lib-openshift.sh]: I think before this line, there could be a short explanation of what this variable means (as it is not described elsewhere). E.g. `OPENSHIFT_CLUSTER_PULLSECRET_PATH variable contains a path to the k8s secret definition that is later used for pulling images from registries that need authentication to access (eg. registry.redhat.io)`

**Comment by hhorak** [comment]
Two suggestions, otherwise LGTM.

File: test-lib-openshift.sh (hunk 1 above)
Comment (pkubatrh) [reply test-lib-openshift.sh]: Makes sense, will fix

File: test-lib-openshift.sh (hunk 2 above)
Comment (pkubatrh) [reply test-lib-openshift.sh]: Good point, will add it to the docstring of the function.

**Comment by pkubatrh** [comment]
Force pushed requested changes

**Comment by hhorak** [comment]
Thanks, merging.

**Pull request merged**